### PR TITLE
Added editing recurring rides

### PR DESCRIPTION
**Pull request by helenxuyang**
### Summary 

This pull request is the first step towards implementing editing single vs all recurring rides

- [x] modified cancel single/all rides modal to reuse for selecting whether to edit single/all rides
- [x] modified create ride modal to show/hide recurring checkbox and info for editing

### Test Plan 

Select whether to edit single/all
image

Edit single
image

Edit all
image

**Edit single occurrences**
Create recurring ride
Edit single non-parent instance
Edit single parent instance

**Edit all occurrences**
Create recurring ride
Edit all occurrences from non-parent ride
Edit all occurrences from parent ride

**Cancel single or all occurrences**
Create recurring ride
Cancel single non-parent instance
Cancel parent instance
Cancel all instances

**Didn't break anything**
Check that creating, editing, and cancelling a single ride still works

### CODE REVIEW

**Comment by mg2239** [comment]
running into a bug where if you click edit all recurring rides, close out, then edit this ride only, the repeating information still shows up

File: frontend/src/components/RequestRideModal/RequestRideInfo.tsx [hunk 1]
```diff
@@ -10,9 +10,12 @@ import CustomRepeatingRides from './CustomRepeatingRides';
 
 type RequestRideInfoProps = {
   ride?: Ride;
+  showRepeatingCheckbox: boolean;
+  showRepeatingInfo: boolean;
 }
 
-const RequestRideInfo = ({ ride }: RequestRideInfoProps) => {
+const RequestRideInfo = ({ ride, showRepeatingCheckbox,
+  showRepeatingInfo }: RequestRideInfoProps) => {
```
Comment (mg2239) [inline frontend/src/components/RequestRideModal/RequestRideInfo.tsx:18]: when would we want the repeating checkbox but not the repeating info (or vice versa)?

Comment (mg2239) [reply frontend/src/components/RequestRideModal/RequestRideInfo.tsx:18]: think this relates to the bug im seeing

Comment (helenxuyang) [reply frontend/src/components/RequestRideModal/RequestRideInfo.tsx:18]: The repeating checkbox only shows up when creating a ride, not when editing a ride, and repeating info shows up for both if it's a repeating ride--I'll look into the bug!

**Review by mg2239 — CHANGES_REQUESTED** [review]
really close! everything seems to work from my testing, except when editing all rides:
- adjusting the custom repeating days doesn't update the repeating days in backend
- after submitting the edits, the master ride changes to unscheduled

**Review by mg2239 — APPROVED** [review]
fixed the state issues and added a couple extra things for safety, like preventing the parent ride start time from moving past it's end date, and not deleting the parent ride when cancelling all repeating rides if it's already occurred 👍

**Pull request merged**